extensions/script-libraries/drawing/drawing.livecodescript

```diff
@@ -0,0 +1,2960 @@
+	return true
+end _svgPathParseSpace
+
+private function _svgPathParseCommaSpace @xPath, @xString
+	if char 1 of xString is comma then
+		delete char 1 of xString
+		return _svgPathParseSpace(xPath, xString)
+	end if
+
+	if not _svgPathParseSpace(xPath, xString) then
+		return false
+	end if
+
+	if char 1 of xString is comma then
+		delete char 1 of xString
+	end if
+
+	return _svgPathParseSpace(xPath, xString)
+end _svgPathParseCommaSpace
+
+private function _svgPathParseCommand @xPath, @xString
+	switch char 1 of xString
+	case "z"
+		return _svgPathParseEmptyCommand(xPath, xString)
+	case "m"
+	case "l"
+	case "t"
+		return _svgPathParseCoordinatePairCommand(xPath, xString, 1)
+	case "h"
+	case "v"
+		return _svgPathParseCoordinateCommand(xPath, xString)
+	case "s"
+	case "q"
+		return _svgPathParseCoordinatePairCommand(xPath, xString, 2)
+	case "c"
+		return _svgPathParseCoordinatePairCommand(xPath, xString, 3)
+	case "a"
+		return _svgPathParseArcCommand(xPath, xString)
+	end switch
+	return false
+end _svgPathParseCommand
+
+private function _svgPathParseEmptyCommand @xPath, @xString
+	local tCommand
+	put char 1 of xString into tCommand
+	delete char 1 of xString
+
+	put tCommand after xPath["commands"]
+	seqPushOntoBack xPath["scalars"], empty
+
+	return _svgPathParseSpace(xPath, xString)
+end _svgPathParseEmptyCommand
+
+private function _svgPathParseCoordinatePairCommand @xPath, @xString, pArity
+	local tCommand
+	put char 1 of xString into tCommand
+	delete char 1 of xString
+
+	local tValid
+	put _svgPathParseSpace(xPath, xString) into tValid
+	repeat while tValid and char 1 of xString is among the chars of "0123456789+-."
+		put tCommand after xPath["commands"]
+		seqPushOntoBack xPath["scalars"], empty
+		repeat pArity times
+			if tValid then put _svgPathParseCoordinatePair(xPath, xString) into tValid
+			if tValid then put _svgPathParseCommaSpace(xPath, xString) into tValid
+		end repeat
+
+		set the caseSensitive to true
+		if tCommand is "m" then
+			put "l" into tCommand
+		else if tCommand is "M" then
+			put "L" into tCommand
+		end if
+		set the caseSensitive to false
+	end repeat
+
+	return tValid
+end _svgPathParseCoordinatePairCommand
+
+private function _svgPathParseCoordinateCommand @xPath, @xString
+	local tCommand
+	put char 1 of xString into tCommand
+	delete char 1 of xString
+
+	local tValid
+	put _svgPathParseSpace(xPath, xString) into tValid
+	repeat while tValid and char 1 of xString is among the chars of "0123456789+-."
+		put tCommand after xPath["commands"]
+		seqPushOntoBack xPath["scalars"], empty
+		if tValid then put _svgPathParseCoordinate(xPath, xString) into tValid
+		if tValid then put _svgPathParseCommaSpace(xPath, xString) into tValid
+	end repeat
+
+	return tValid
+end _svgPathParseCoordinateCommand
+
+private function _svgPathParseArcCommand @xPath, @xString
+	local tCommand
+	put char 1 of xString into tCommand
+	delete char 1 of xString
+
+	local tValid
+	put _svgPathParseSpace(xPath, xStri
```

Comment:
The indentation here is a bit confusing!

Reply:
In what way?

Reply:
oh! this has moved to a different place